org.godotengine.Godot3.yaml

```diff
@@ -0,0 +1,133 @@
+    SCONS_FLAGS: >
+      platform=linuxbsd
+      CCFLAGS=-I/app/include
+      prefix=/app
+      unix_global_settings_path=/app
+      progress=no
+      builtin_freetype=no
+      builtin_graphite=no
+      builtin_harfbuzz=no
+      builtin_libogg=no
+      builtin_libpng=no
+      builtin_libtheora=no
+      builtin_libvorbis=no
+      builtin_libwebp=no
+      builtin_openssl=no
+      builtin_libvpx=no
+      builtin_zlib=no
+      udev=no
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=host
+  - --device=all
+  - --talk-name=org.freedesktop.Flatpak
+
+modules:
+  - shared-modules/glu/glu-9.json
+
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk11/install.sh
+
+  - name: dotnet
+    buildsystem: simple
+    build-options:
+        arch:
+            x86_64:
+                env:
+                    RUNTIME: linux-x64
+    build-commands:
+      - /usr/lib/sdk/dotnet7/bin/install-sdk.sh
+      - cp -r /usr/lib/sdk/dotnet7/lib/packs/ /app/lib/dotnet/
+
+  - name: scons
+    buildsystem: simple
+    cleanup: ['*']
+    sources:
+      - type: archive
+        sha256: 3d43b2303a924816ea0e1b345ff04c9b3e27b53eadf0f26012fc0c29b019685f
+        url: https://downloads.sourceforge.net/project/scons/scons/4.4.0/SCons-4.4.0.tar.gz
+    build-commands:
+      - pip3 install --no-index --no-build-isolation --prefix=/app .
+
+  - name: godot-tools
+    buildsystem: simple
+    sources:
+      - type: archive
+        sha256: 3cb48126b76858f40cf54bd345bb84dc1f49d9e6f8a4a7425ad86e805d39701d
+        url: https://downloads.tuxfamily.org/godotengine/3.5.3/godot-3.5.3-stable.tar.xz
+
+      - type: script
+        dest-filename: godot.sh
+        commands:
+          - export APPDATA="$XDG_DATA_HOME"
+          - export PATH="/app/jre/bin:$PATH"
+          - export DOTNET_ROOT="/app/lib/dotnet"
+          - export DOTNET_CLI_TELEMETRY_OPTOUT=true
+          - export DOTNET_NOLOGO=true
+          - /app/bin/godot-bin "$@"
+      - type: file
+        path: nuget/nuget-source.config
+      - type: file
+        url: https://www.nuget.org/api/v2/package/Microsoft.Build.NoTargets/2.0.1
+        sha512: 19fd7d616993d12d633674fe1e9779eab0aa40aa21eec412561e28e8a7e2de3bae8d9fef9595257134d73f33a9dea9ee0e4ebbafbd32525c123320c1eb5bf160
+        dest: nuget-sources
+        dest-filename: microsoft.build.notargets.2.0.1.nupkg
+      - nuget/nuget-sources.json
+      - type: patch
+        path: update-appdata.patch
+
+    build-commands:
+      - cp nuget-source.config modules/mono/glue/GodotSharp/nuget.config
+      - cp nuget-source.config modules/mono/editor/GodotTools/nuget.config
+      - cp nuget-source.config modules/mono/editor/Godot.NET.Sdk/nuget.config
+      - python3 /app/bin/scons $SCONS_FLAGS $SCONS_FLAGS_EXTRA target=editor precision=single module_mono_enabled=yes -j "$FLATPAK_BUILDER_N_JOBS"
```

Comment:
with `release_debug`:

```
RuntimeError: Mono glue sources not found. Did you forget to run '--generate-mono-glue'?:
  File "/run/build/godot-tools/SConstruct", line 719:
    SConscript("modules/SCsub")
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/run/build/godot-tools/modules/SCsub", line 21:
    SConscript(name + "/SCsub")  # Built-in.
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/run/build/godot-tools/modules/mono/SCsub", line 24:
    raise RuntimeError("Mono glue sources not found. Did you forget to run '--generate-mono-glue'?")
Error: module godot-tools: Child process exited with code 2
```

Reply:
The changes here should take care of this: https://github.com/flathub/flathub/pull/4595/commits/b0c8e8dbf67958933cd4825a7a68273c119c5ace https://github.com/flathub/flathub/pull/4595/commits/44e22bd40dd0248c3524345b0a026e45e5be6952 https://github.com/flathub/flathub/pull/4595/commits/4209392840016eee579bc14121921ef40c41dd82

Reply:
> The changes here should take care of this: [b0c8e8d](https://github.com/flathub/flathub/commit/b0c8e8dbf67958933cd4825a7a68273c119c5ace) [44e22bd](https://github.com/flathub/flathub/commit/44e22bd40dd0248c3524345b0a026e45e5be6952) [4209392](https://github.com/flathub/flathub/commit/4209392840016eee579bc14121921ef40c41dd82)

Still fails

<details>
<summary> Log <summary>

```
 -j "$FLATPAK_BUILDER_N_JOBS"
scons: Reading SConscript files ...
Note: Building a debug binary (which will run slowly). Use `target=release_debug` to build an optimized release binary.
Using custom build name: flathub
RuntimeError: Mono glue sources not found. Did you forget to run '--generate-mono-glue'?:
  File "/run/build/godot-tools/SConstruct", line 719:
    SConscript("modules/SCsub")
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/run/build/godot-tools/modules/SCsub", line 21:
    SConscript(name + "/SCsub")  # Built-in.
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/app/lib/python3.11/site-packages/SCons/Script/SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/run/build/godot-tools/modules/mono/SCsub", line 24:
    raise RuntimeError("Mono glue sources not found. Did you forget to run '--generate-mono-glue'?")
Error: module godot-tools: Child process exited with code 2
```

</details>

Reply:
I'm trying to follow the instructions set here: https://docs.godotengine.org/en/3.5/development/compiling/compiling_with_mono.html#generate-the-glue

![image](https://github.com/flathub/flathub/assets/45298929/7d6194a8-128e-4dd2-9596-cf5efd58c1c5)

The build instructions say to build Godot first, then run the binary to generate the Mono glue code. _Then_, we need to rebuild Godot with the generated glue code: https://docs.godotengine.org/en/3.5/development/compiling/compiling_with_mono.html#rebuild-with-mono-glue

![image](https://github.com/flathub/flathub/assets/45298929/044cfd18-15b0-4d04-aa7a-00ece3a054a6)

I'm assuming [this line](https://github.com/flathub/flathub/blob/4209392840016eee579bc14121921ef40c41dd82/org.godotengine.Godot3Sharp.yaml#L106) is where the build fails on your end, since [I specified the `release_debug` target two lines after it](https://github.com/flathub/flathub/blob/4209392840016eee579bc14121921ef40c41dd82/org.godotengine.Godot3Sharp.yaml#L108). If my assumption's correct, this means the build's failing [_before the glue code can even be generated_](https://github.com/flathub/flathub/blob/4209392840016eee579bc14121921ef40c41dd82/org.godotengine.Godot3Sharp.yaml#L107). According to the docs, this shouldn't be happening. I'll need to look into this further.